app/models/spree/calculator/variable_percent_on_line_item.rb

```diff
@@ -0,0 +1,20 @@
+class Spree::Calculator::VariablePercentOnLineItem < Spree::Calculator
+  preference :minimum_discount, :decimal, default: 0
+  preference :minimum_discount_on_price_greater_than, :decimal, default: 0
+  preference :maximum_discount_on_price_greater_than, :decimal, default: 0
+  preference :maximum_discount, :decimal, default: 0
+  def self.description
+    "Calculator for discount on line item"
+  end
+  def compute(line_item)
+    price = line_item.price
+    if price > preferred_maximum_discount_on_price_greater_than
```

Comment:
@rishabhguptarishi what about the case when the price is equal to preferred_maximum_discount_on_price_greater_than?

Reply:
according to the requirement discount will only be applicable if price > preferred_maximum_discount_on_price_greater_than so if price is equal to it discount wont be applicable.

Reply:
@rishabhguptarishi please add equal_to condition in the requirement as well and make the change at both places.